streaming/src/main/scala/org/apache/spark/streaming/util/FileBasedWriteAheadLog.scala

```diff
@@ -251,4 +261,23 @@ private[streaming] object FileBasedWriteAheadLog {
       }
     }.sortBy { _.startTime }
   }
+
+  /**
+   * This creates an iterator from a parallel collection, by keeping at most `n` objects in memory
+   * at any given time, where `n` is the size of the thread pool. This is crucial for use cases
+   * where we create `FileBasedWriteAheadLogReader`s during parallel recovery. We don't want to
+   * open up `k` streams altogether where `k` is the size of the Seq that we want to parallelize.
+   */
+  def seqToParIterator[I, O](
+      tpool: ThreadPoolExecutor,
+      source: Seq[I],
+      handler: I => Iterator[O]): Iterator[O] = {
+    val taskSupport = new ThreadPoolTaskSupport(tpool)
+    val groupSize = math.max(math.max(tpool.getCorePoolSize, tpool.getPoolSize), 8)
```

Comment:
And you need to `ThreadUtils.newDaemonCachedThreadPool(...,maxThreadNumber)` to set it. Actually, I think you can just set the `maxThreadNumber` to 8 and use it here.